### ISSUE TITLE
ifc file parsing failed!

### ISSUE THREAD

**Issue by JickYang** [IfcOpenShell/IfcOpenShell#270]
When I use Java IfcOpenShellPlugin parse a ifc file , It return 255 code. And The version is IfcOpenShell-0.5.0-dev-2. Can you tell me why?

The ifc file is [out.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/1382240/out.zip)

**Comment by aothms** [comment]
Hi, can you provide a bit more details? Thanks.

> Java IfcOpenShellPlugin

I assume you are referring to https://github.com/opensourceBIM/IfcOpenShell-BIMserver-plugin/

> It return 255 code

What function when and where returns that?

**Comment by JickYang** [comment]
@aothms Thanks for answer this.
Yes, we use the 'IfcOpenShell-BIMserver-plugin' in our project.  When it parse that ifc file, the program displays the following log.
```
org.bimserver.plugins.renderengine.RenderEngineException: Exited with non-zero exit code: 255
```
I want to know what dose this code mean?

**Comment by aothms** [comment]
I cannot reproduce this issue. There is a non-trivial interaction between BIMserver and IfcOpenShell (there are quite some optimizations that the BIMserver does before sending the file to IfcOpenShell). Below is code to run the geometry generation process isolated from the BIMserver, which does not seem to result in any errors for the file you provided. It is probably best to involve @rubendel who is much more knowledgeable on the interactions within the entire system. Perhaps an issue here would be more suitable: https://github.com/opensourceBIM/IfcOpenShell-BIMserver-plugin/ and provide more details on your environment and log output generated during the process.

~~~java
package org.ifcopenshell;

import java.io.FileInputStream;
import java.io.FileNotFoundException;

import org.bimserver.plugins.renderengine.RenderEngineException;

public class Runner {
	
	public static void main(String [] args)
	{
		IfcGeomServerClient client;
		
		try {
			client = new IfcGeomServerClient(args[0]);
		} catch (RenderEngineException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return;
		}
		
		try {
			client.loadModel(new FileInputStream(args[1]));
		} catch (FileNotFoundException | RenderEngineException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return;
		}

		while (client.hasNext()) {
			try {
				IfcGeomServerClientEntity next = client.getNext();
				if (next == null) {
					System.out.println("Internal error");
					return;
				}
				System.out.println(String.format("%s %s", next.getType(), next.getGuid()));
				float area = next.getExtendedDataAsFloat("TOTAL_SURFACE_AREA");
				float volume = next.getExtendedDataAsFloat("TOTAL_SHAPE_VOLUME");
				if (next.getType().equals("IfcSpace")) {
					float walkable = next.getExtendedDataAsFloat("WALKABLE_SURFACE_AREA");
					System.out.println(String.format("Volume: %.2f; Area: %.2f; Walkable %.2f", volume, area, walkable));
				} else {
					System.out.println(String.format("Volume: %.2f; Area: %.2f", volume, area));
				}
			} catch (RenderEngineException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
				return;
			}
		}
		
		try {
			client.close();
		} catch (RenderEngineException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		
		System.exit(0); 
	}
	
}
~~~

**Comment by rubendel** [comment]
BIMserver does indeed split the original file into smaller files that can be processed in parallel, keeping the memory usage under control and giving an idea of the progress.

For any file that generates a non-zero exit-code, the (sub) IFC file is stored under "home/debug/[datetime]/type[i].ifc".

For this particular model, there was 1 bug in BIMserver that I have fixed (will be in 1.5.89). In some cases, objects of type "IfcFacetedBrep" were not included in the sub-ifc-files.

After fixing that bug, the model does load successfully, but there are still 3 (similar) files that are not processed correctly by IfcOpenShell. I attached one of them to this issue. The exit-code I am getting is "-1073741819" which looks a bit weird.

**Comment by aothms** [comment]
Many thanks Ruben, there does appear to be some invalid geometry in there, and the way this is handled apparently depends on the version of Open CASCADE, which is why I missed it in the original file.

This geometry has one segment missing:
~~~
#464=IfcCompositeCurve((#370,#368,#367,#365),.F.)
#370=IfcCompositeCurveSegment(.CONTINUOUS.,.T.,#202)
#202=IfcPolyline((#51,#50,#49,#48))
#51=IfcCartesianPoint((1145.,0.))
#50=IfcCartesianPoint((1145.,1780.))
#49=IfcCartesianPoint((255.,1780.))
#48=IfcCartesianPoint((-132.051142581968,1439.7352592686))
#368=IfcCompositeCurveSegment(.CONTINUOUS.,.T.,#369)
#369=IfcTrimmedCurve(#248,(IfcParameterValue(0.)),(IfcParameterValue(0.849637138838738)),.T.,.PARAMETER.)
#248=IfcCircle(#214,200.)
#214=IfcAxis2Placement2D(#47,#215)
#47=IfcCartesianPoint((-2.27373675443232E-013,1289.52708458161))
#215=IfcDirection((-0.660255712909837,0.751040873434944))
#367=IfcCompositeCurveSegment(.CONTINUOUS.,.T.,#201)
#201=IfcPolyline((#46,#45))
#46=IfcCartesianPoint((-200.,1289.52708458161))
#45=IfcCartesianPoint((-200.,200.))
#365=IfcCompositeCurveSegment(.DISCONTINUOUS.,.T.,#366)
#366=IfcTrimmedCurve(#247,(IfcParameterValue(0.)),(IfcParameterValue(1.5707963267949)),.T.,.PARAMETER.)
#247=IfcCircle(#212,200.)
#212=IfcAxis2Placement2D(#44,#213)
#44=IfcCartesianPoint((5.6843418860808E-014,200.))
#213=IfcDirection((-1.,0.))
~~~

![ifcdiscreteaccessory-0 svg](https://user-images.githubusercontent.com/1096535/32062045-9f616316-ba73-11e7-9d0f-caccaf184a51.png)

This apparently causes issues later on when the shape is used in boolean ops.

I created a new issue for that.
